crates/ra_syntax/src/grammar/expressions.rs

```diff
@@ -213,7 +213,7 @@ const LHS_FIRST: TokenSet = token_set_union![
     atom::ATOM_EXPR_FIRST,
 ];
 
-fn lhs(p: &mut Parser, r: Restrictions) -> Option<CompletedMarker> {
+fn lhs(p: &mut Parser, r: Restrictions) -> (Option<CompletedMarker>, Option<BlockLike>) {
```

Comment:
I would expect an `Option<(CompletionMarker, BlockLike)>` here?

Reply:
I currently only give a `BlockLike` in case of macro invocation. This code makes that work without breaking anything else.